dask/array/core.py

```diff
@@ -378,6 +378,10 @@ def rec_concatenate(arrays, axis=0):
         arrays = [a[None, ...] for a in arrays]
     if len(arrays) == 1:
         return arrays[0]
+    for a in arrays:
+        if hasattr(a, 'shape'):
+            if all(i == 0 for i in a.shape):
+                arrays.remove(a)
```

Comment:
I'm still confused about why the list comprehension above isn't equivalent
to this code.  I'm not raising concern about the removal of empty arrays,
I'm just suggesting an alternative way to write this code.

On Mon, May 4, 2015 at 11:48 AM, Blake Griffith <notifications@github.com>
wrote:

> In dask/array/core.py
> <https://github.com/ContinuumIO/dask/pull/176#discussion_r29611406>:
>
> > @@ -378,6 +378,10 @@ def rec_concatenate(arrays, axis=0):
> >          arrays = [a[None, ...] for a in arrays]
> >      if len(arrays) == 1:
> >          return arrays[0]
> > +    for a in arrays:
> > +        if hasattr(a, 'shape'):
> > +            if all(i == 0 for i in a.shape):
> > +                arrays.remove(a)
>
> @mrocklin <https://github.com/mrocklin> I think handling the shapes like
> this here is a symptom of still not handling the ghost slices right. I just
> added some new tests to show this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/176/files#r29611406>.
>